Lib/test/test_asyncio/test_subprocess.py

```diff
@@ -699,34 +843,40 @@ class SubprocessPidfdWatcherTests(SubprocessWatcherMixin,
                                       test_utils.TestCase):
         Watcher = unix_events.PidfdChildWatcher
 
-else:
-    # Windows
-    class SubprocessProactorTests(SubprocessMixin, test_utils.TestCase):
-
-        def setUp(self):
-            super().setUp()
-            self.loop = asyncio.ProactorEventLoop()
-            self.set_event_loop(self.loop)
-
 
-class GenericWatcherTests:
+    class GenericWatcherTests(test_utils.TestCase):
 
-    def test_create_subprocess_fails_with_inactive_watcher(self):
+        def test_create_subprocess_fails_with_inactive_watcher(self):
+            watcher = mock.create_autospec(
+                asyncio.AbstractChildWatcher,
+                **{"__enter__.return_value.is_active.return_value": False}
+            )
 
-        async def execute():
-            watcher = mock.create_authspec(asyncio.AbstractChildWatcher)
-            watcher.is_active.return_value = False
-            asyncio.set_child_watcher(watcher)
+            async def execute():
+                asyncio.set_child_watcher(watcher)
 
-            with self.assertRaises(RuntimeError):
-                await subprocess.create_subprocess_exec(
-                    os_helper.FakePath(sys.executable), '-c', 'pass')
+                with self.assertRaises(RuntimeError):
+                    await subprocess.create_subprocess_exec(
+                        os_helper.FakePath(sys.executable), '-c', 'pass')
 
-            watcher.add_child_handler.assert_not_called()
+                watcher.add_child_handler.assert_not_called()
 
-        self.assertIsNone(self.loop.run_until_complete(execute()))
+            with Runner(loop_factory=asyncio.new_event_loop) as runner:
+                self.assertIsNone(runner.run(execute()))
```

Comment:
Why not just use `self.loop.run_until_complete()` as in other tests?

Reply:
the `GenericWatcherTests` don't have a `self.loop`, and `SubprocessWatcherMixin` (which sets `self.loop`) calls `get_event_loop_policy().attach_loop` and this test case is testing what happens when `attach_loop` isn't called

Reply:
Many classes set `self.loop`. For this test you can inherit from `test_utils.TestCase` and add
```python
        def setUp(self):
            super().setUp()
            self.loop = asyncio.new_event_loop()
            self.set_event_loop(self.loop)
```